### PR TITLE
Bug 1185063 - Use AppWindow's getSiteIconUrl in task manager cards

### PR DESCRIPTION
**Pull request by sfoster**
Changes to Card, plus removing now-obsolete code in CardsHelper. The poor cards_helper unit test had no test for getOffOrigin which left it empty. We'll want to finish removing this in another bug so I backfilled a quick test to stop it being entirely empty. 

I added small-icon support in here, but also noted that we're still getting the smallest 16x16 icon in a favicon.ico that has larger sizes available in it. However, it makes it easy to test :/

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4dd65e7889bdac2f9d0ec7e951f471bd3613fc5c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b0268ddce558b769933950f0acf7ddc1c90d0743

File: apps/system/js/card.js [hunk 1]
```diff
@@ -477,6 +495,18 @@
     return displayString;
   };
 
+  function debug(message) {
+    if (Card.DEBUG) {
```
Comment (etiennesegonzac) [inline apps/system/js/card.js]: nit: early return

File: apps/system/test/marionette/task_manager_icons_test.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  // we can map from a color to a particular size icon image
+  var iconColorToSize = {
+    '255,255,0,255': 16,
+    '0,255,255,255': 32,
+    '255,0,255,255': 64,
+    '0,0,255,255': 128,
+    '255,0,0,255': 256
+  };
+
+  function getElementColor(elem, expectedSize) {
```
Comment (etiennesegonzac) [inline apps/system/test/marionette/task_manager_icons_test.js]: badass :)
Can you file a bug to investigate using this to cover the screenshoting in the card view?

File: apps/system/style/cards_view/cards_view.css [hunk 3]
```diff
@@ -262,13 +262,16 @@
...
-
 }
 #cards-view .card.browser .card-tray > button.appIcon {
   background-image: url('../../shared/resources/branding/browser_126.png');
 }
+#cards-view .card .card-tray > button.appIcon.small-icon {
+  background-color: #CFCFCF;
+  background-size: 1.6rem auto;
+  border-radius: 100%;
+}
```
Comment (etiennesegonzac) [inline apps/system/style/cards_view/cards_view.css]: do we want to add basic css for the `.pending` case?

File: apps/system/js/card.js [hunk 4]
```diff
@@ -288,60 +283,83 @@
...
+    var cachedLayer = isActive && app.requestScreenshotURL();
+    screenshotView.classList.toggle('fullscreen',
+                                    isActive && app.isFullScreen());
+    screenshotView.classList.toggle('maximized',
+                                  isActive && app.appChrome &&
+                                  app.appChrome.isMaximized());
     screenshotView.style.backgroundImage =
       (cachedLayer ? 'url(' + cachedLayer + ')' : 'none' ) + ',' +
       '-moz-element(#' + this.app.instanceID + ')';
+      '-moz-element(#' + app.instanceID + ')';
```
Comment (etiennesegonzac) [inline apps/system/js/card.js]: ? missing line deletion?

File: apps/system/test/unit/card_test.js [hunk 5]
```diff
@@ -228,9 +265,30 @@ suite('system/Card', function() {
         manager: mockManager
       });
       appCard.render();
+      getIconPromise.mFulfillToValue({ url: iconDataURI });
```
Comment (etiennesegonzac) [inline apps/system/test/unit/card_test.js]: can we move some of those calls to a `teardown` ?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8c556754005880632aaacbdc5cda35ff8a8072d0
